backend/reg.mli

```diff
@@ -75,12 +75,18 @@ and stack_location =
 val equal_location : location -> location -> bool
 
 val dummy: t
+(* CR gyorsh for xclerc: is there a way around it? *)
+val dummy_for_regalloc: t
```

Comment:
I think the only use is for `ArraySet`,
which could arguably be rewritten
using `DynArray` (the latest upstream
version, not the one we currently
have in this repository), which is
based on its own non-domain-specific
notion of a dummy...